ra-optimisation/linear-range-action-rao/src/main/java/com/farao_community/farao/linear_range_action_rao/LinearRangeActionRao.java

```diff
@@ -35,7 +51,60 @@ public String getVersion() {
 
     @Override
     public CompletableFuture<RaoComputationResult> run(Network network, Crac crac, String variantId, ComputationManager computationManager, RaoParameters parameters) {
-        // TODO : implement LinearRangeActionRao
         return null;
     }
+
+    public CompletableFuture<RaoComputationResult> run(Network network, Crac crac, String variantId,
+                                                       ComputationManager computationManager, RaoParameters parameters,
+                                                       SensitivityComputationFactory sensitivityComputationFactory) {
+        SensitivityComputationService.init(sensitivityComputationFactory, computationManager);
+        SensitivitySecurityAnalysisResult sensiSaResults = SensitivitySecurityAnalysisService.runSensitivity(network, crac, computationManager);
+
+        SensitivityComputationResults preSensi = sensiSaResults.getPrecontingencyResult();
+        PreContingencyResult preRao = new PreContingencyResult();
+        // 1. do for pre
+        // get cnec from crac
+        crac.getCnecs().forEach(cnec -> {
+            LOGGER.info("Cnec: " + cnec.getId());
+            preSensi.getSensitivityValues().forEach(sensitivityValue -> {
+                String id = sensitivityValue.getFactor().getFunction().getId();
+                String name = sensitivityValue.getFactor().getFunction().getName();
+                double maximumFlow = Integer.MAX_VALUE; //todo ? how to get max from Crac?
+                double preOptimisationFlow = sensitivityValue.getFunctionReference();
+                MonitoredBranchResult monitoredBranchResult = new MonitoredBranchResult(id, name, id, maximumFlow, preOptimisationFlow, preOptimisationFlow);
+                preRao.addMonitoredBranchResult(monitoredBranchResult);
+                LOGGER.info("ID: " + id + "; preOptimisationFlow = " + preOptimisationFlow);
+            });
```

Comment:
You can extract this as a method so you can reuse it after contingencies

Reply:
ok